updater/lib/dependabot/api_client.rb

```diff
@@ -274,8 +276,76 @@ def increment_metric(metric, tags:)
       end
     end
 
+    sig { params(ecosystem: Ecosystem).void }
+    def record_ecosystem_meta(ecosystem)
+      return unless Dependabot::Experiments.enabled?(:record_ecosystem_meta)
+
+      return if ecosystem.nil?
+
+      begin
+        ::Dependabot::OpenTelemetry.tracer.in_span("record_ecosystem_meta", kind: :internal) do |_span|
+          api_url = "#{base_url}/update_jobs/#{job_id}/record_ecosystem_meta"
+
+          body = {
+            data: {
+              ecosystem: {
+                name: ecosystem.name,
+                package_manager: version_manager_json(ecosystem.package_manager),
+                language: version_manager_json(ecosystem.language)
+              }
+            }
+          }
+
+          retry_count = 0
+
+          begin
+            response = http_client.post(api_url, json: body)
+            raise ApiError, response.body if response.code >= 400
+          rescue HTTP::ConnectionError, OpenSSL::SSL::SSLError, ApiError => e
+            retry_count += 1
+            if retry_count <= MAX_REQUEST_RETRIES
+              sleep(rand(3.0..10.0))
+              retry
+            else
+              Dependabot.logger.error(
+                "Failed to record ecosystem meta after #{MAX_REQUEST_RETRIES} retries: #{e.message}"
+              )
+            end
+          end
+        end
+      rescue StandardError => e
+        Dependabot.logger.error("Failed to record ecosystem meta: #{e.message}")
+      end
+    end
+
     private
 
+    sig { params(version_manager: T.nilable(Dependabot::Ecosystem::VersionManager)).void }
+    def version_manager_json(version_manager)
+      return unless version_manager
+
+      {
+        name: version_manager.name,
+        raw_version: version_manager.version.to_semver.to_s,
+        version: version_manager.version.to_s,
+        requirement: version_manager_requirement_json(version_manager)
+      }
+    end
+
+    sig { params(version_manager: Dependabot::Ecosystem::VersionManager).void }
+    def version_manager_requirement_json(version_manager)
+      requirement = version_manager.requirement
+      return unless requirement
+
+      {
+        raw_constraint: requirement.constraints.join(", "),
+        min_raw_version: requirement.min_version&.to_semver.to_s,
+        min_version: requirement.min_version&.to_s,
+        max_raw_version: requirement.max_version&.to_semver.to_s,
+        max_version: requirement.max_version&.to_s
+      }
+    end
+
```

Comment:
**Review Tip:** Use common metrics creation functions for package manager and language since they have metrics in same structure.